File: app/src/main/java/com/xht/androidnote/module/eventdispatch/NestedScrollViewTestActivity.java
package com.xht.androidnote.module.eventdispatch;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentActivity;
import androidx.viewpager2.adapter.FragmentStateAdapter;
import androidx.viewpager2.widget.ViewPager2;

import com.google.android.material.tabs.TabLayout;
import com.google.android.material.tabs.TabLayoutMediator;
import com.xht.androidnote.R;

public class NestedScrollViewTestActivity extends AppCompatActivity {

    private ViewPager2 viewPager;
    private TabLayout tabLayout;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_nested_scroll_view_test);

        Toolbar toolbar = findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        viewPager = findViewById(R.id.viewPager);
        tabLayout = findViewById(R.id.tabLayout);

        viewPager.setAdapter(new ViewPagerAdapter(this));

        new TabLayoutMediator(tabLayout, viewPager,
                (tab, position) -> tab.setText("Tab " + (position + 1))
        ).attach();
    }

    private class ViewPagerAdapter extends FragmentStateAdapter {
        public ViewPagerAdapter(FragmentActivity fragmentActivity) {
            super(fragmentActivity);
        }

        @Override
        public Fragment createFragment(int position) {
            return new ScrollableFragment();
        }

        @Override
        public int getItemCount() {
            return 3;
        }
    }
}

File: app/src/main/java/com/xht/androidnote/module/eventdispatch/NestedScrollViewTestActivity.kt
package com.xht.androidnote.module.eventdispatch

import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.xht.androidnote.R
import com.xht.androidnote.base.BaseViewActivity
import com.xht.androidnote.databinding.ActivityNestedScrollViewTestBinding

class NestedScrollViewTestActivity : BaseViewActivity<ActivityNestedScrollViewTestBinding>() {
    override fun getViewBinding(): ActivityNestedScrollViewTestBinding {
        return ActivityNestedScrollViewTestBinding.inflate(layoutInflater)
    }

    override fun getLayoutId(): Int {
        return R.layout.activity_nested_scroll_view_test
    }

    override fun initEventAndData() {

        val recyclerView: RecyclerView = findViewById(R.id.recyclerView)
        recyclerView.layoutManager = LinearLayoutManager(this)

        // 生成一些测试数据
        val itemList = List(20) { Person("Item #$it") }

        // 绑定 Adapter
        val adapter = MyAdapter(itemList)
        recyclerView.adapter = adapter
    }

}

data class Person(val name: String)
